File: Array/maximunSubarray.py
 def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) = 1 
           return 1
        
        curr = nums[0]
        maxsofar = nums[0]
        
        for num in nums:
            curr = max(num, curr+num)
            if curr > maxsofar:
                maxsofar = curr
        return maxsofar

File: Array/maximunSubarray.js
var maxSubArray = function(nums) {
    let curr = nums[0]
    let global = nums[0]
    
    for(let i=1; i<nums.length; i++){
        curr = Math.max(nums[i], curr+nums[i])
        
        if(curr > global){
            global = curr
        }
    }
    return global
};
